libbeat/docs/shared-cloudfoundry.asciidoc

```diff
@@ -0,0 +1,128 @@
+
+
+[float]
+==== Cloud Foundry deploy manifests
+
+You deploy {beatname_uc} as an application with no route.
+
+Cloud Foundry requires that 3 files exist inside of a directory to allow {beatname_uc} to be pushed. The commands
+below provide the basic steps for getting it up and running.
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+curl -L -O https://artifacts.elastic.co/downloads/beats/{beatname_lc}/{beatname_lc}-{version}-linux-x86_64.tar.gz
+tar xzvf {beatname_lc}-{version}-linux-x86_64.tar.gz
+cd {beatname_lc}-{version}-linux-x86_64
+curl -L -O https://raw.githubusercontent.com/elastic/beats/{branch}/deploy/cloudfoundry/{beatname_lc}/{beatname_lc}.yml
+# Update api_address, client_id, client_secret in downloaded {beatname_lc}.
+curl -L -O https://raw.githubusercontent.com/elastic/beats/{branch}/deploy/cloudfoundry/{beatname_lc}/manifest.yml
+------------------------------------------------
+
+
+[float]
+==== Deploy
+
+To deploy {beatname_uc} to Cloud Foundry, run:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+cf push
+------------------------------------------------
+
+To check the status, run:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+$ cf apps
+
+name       requested state   instances   memory   disk   urls
+{beatname_lc}   started           1/1         512M     1G
+------------------------------------------------
+
+ifeval::["{beatname_lc}"=="filebeat"]
+Log events should start flowing to Elasticsearch.
+endif::[]
+ifeval::["{beatname_lc}"=="metricbeat"]
+Metric events should start flowing to Elasticsearch.
+endif::[]
+The events are annotated with metadata added by the <<add-cloudfoundry-metadata>> processor.
+
+[float]
+=== Scaling {beatname_uc}
+
+A single instance of {beatname_uc} is able to ship more than 100 thousand events
+per minute. If your Cloud Foundry deployment is producing more events than
+{beatname_uc} can collect and ship, the Firehose will start dropping events and it
+will mark {beatname_uc} as slow consumer. If problems persist, {beatname_uc} may
+be disconnected from the Firehose.
+In these cases, you will need to scale {beatname_uc} to avoid losing events, find
+here some guidelines about how to do it.
+
+The main settings you need to take into account are:
+
+ifeval::["{beatname_lc}"=="filebeat"]
+* The `shard_id` specified in the `cloudfoundry` input configuration. The
+  firehose will divide the events between all the {beatname_uc} instances with
+  the same value for this setting. All the instances with the same `shard_id`
+  should have the same configuration.
+endif::[]
+ifeval::["{beatname_lc}"=="metricbeat"]
+* The `shard_id` specified in the `cloudfoundry` module. The
+  firehose will divide the events between all the {beatname_uc} instances with
+  the same value for this setting. All the instances with the same `shard_id`
+  should have the same configuration
```

Comment:
```suggestion
  the same value for this setting. All instances with the same `shard_id`
  should have the same configuration.
```

Reply:
I think by "same configuration" you mean same configuration for things like processors, etc., right? Should we clarify this here?